kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/encryption/EnvelopeEncryptionFilterIT.java

```diff
@@ -248,6 +257,78 @@ void produceAndConsumeEncryptedAndPlainTopicsAtSameTime(KafkaCluster cluster, To
         }
     }
 
+    /**
+     * Test that ensures that the record offsets returned by the broker are faithfully relayed to the client.
+     * @param cluster underlying kafka cluster
+     * @param compactedTopic topic configured for compaction.
+     * @param directConsumer consumer connected directly to the underlying kafka cluster
+     * @param testKmsFacade kms facade
+     * @throws Exception exception
+     */
+    @TestTemplate
+    @SuppressWarnings("java:S2925")
+    void offsetFidelity(@BrokerConfig(name = "log.cleaner.backoff.ms", value = "50") KafkaCluster cluster,
+                        @TopicConfig(name = "segment.ms", value = "125") @TopicConfig(name = "cleanup.policy", value = "compact") Topic compactedTopic,
```

Comment:
I get that `compact` is the more complicated policy and thus the one we need to ensure fidelity with, but should we also have a version of the test with a non compact topic?

Reply:
Yes, I agree there's scope for more testing.  One case that's not covered is the case where a topic's offset begin at a > 0 value on the broker (i.e. after retention rules have caused a segment to be deleted).  I think are rather leave this kind of thing for another PR.

Reply:
Yeah happy to leave for a new PR